backend/cmm_helpers.ml

```diff
@@ -749,35 +751,46 @@ let mod_int c1 c2 is_safe dbg =
 (* Division or modulo on boxed integers. The overflow case min_int / -1 can
    occur, in which case we force x / -1 = -x and x mod -1 = 0. (PR#5513). *)
 
-let is_different_from x = function
-  | Cconst_int (n, _) -> n <> x
-  | Cconst_natint (n, _) -> n <> Nativeint.of_int x
-  | _ -> false
+(* Division or modulo on boxed integers. The overflow case min_int / -1 can
+   occur, in which case we force x / -1 = -x and x mod -1 = 0. (PR#5513). *)
```

Comment:
maybe add in a comment that the specified behavior is reached via the `mkop` and `mkm1` functions, which are used to build the correct cmm expressions respectively  in the usual case, and in the case where the divisor is `-1` (and the arch crashes on overflow).